### PR TITLE
Fix1558

### PR DESCRIPTION
**Pull request by Dalimil**
Fixes Issue: #1558 
Component state is a bit complicated to maintain because 'indeterminate' can only be sent via JavaScript and 'checked=true' seems to override 'indeterminate=true'. So I created an uncontrolled input that is initialized everytime it mounts...
I have one concern though, because toggleAllBreakpoints simply modifies the state by toggling breakpoints one-by-one, the state of our checkbox toggle will reflect that and when a user clicks disableAll, there will be a short time period when the checkbox is indeterminate (just before disabling the last breakpoint) so that's why it flickers and looks a bit unsteady.  
The same thing happens whenever breakpoints are loading - this if conditional was there before (with the original switch button `if (breakpoints.size == 0 or breakpointsLoading)`) but it might also cause the flicker now... We could probably remove it and just have this conditional for the input's `disabled` property. 

Need feedback...

@jasonLaster 

out

### CODE REVIEW

**Comment by jasonLaster** [comment]
Great GIF! Looks awesome. 

@clarkbw thoughts?

**Comment by jasonLaster** [comment]
I don't have a great solution for the flickering. We can brainstorm later

**Comment by wldcordeiro** [comment]
This needs a rebase on top of the changes for vertical layout. @Dalimil

File: src/components/SecondaryPanes.js [hunk 1]
```diff
@@ -45,6 +51,39 @@ const SecondaryPanes = React.createClass({
...
+      className: boxClassName,
+      onClick: () => toggleAllBreakpoints(!breakpointsDisabled),
+      defaultChecked: true,
+      ref: (input) => {
+        if (input != null) {
+          const isIndeterminate = !breakpointsDisabled &&
+            breakpoints.some(x => x.disabled);
+          input.indeterminate = isIndeterminate;
+          // checked=true would override the indeterminate value
+          input.checked = !breakpointsDisabled && !isIndeterminate;
```
Comment (jasonLaster) [inline src/components/SecondaryPanes.js]: why can't these be normal props?

Comment (nt1m) [inline src/components/SecondaryPanes.js]: Was going to say the same thing  :) They both are standard dom properties, so it should just work as normal props

Comment (Dalimil) [inline src/components/SecondaryPanes.js]: From what I know about the indeterminate property, it isn't possible to set it in HTML - only in JavaScript.
`checked` on the other hand could be a standard prop as you mention but it depends on the indeterminate status (we don't want to override the value by setting checked to true). But you're right that `checked` could be set as a prop - but we would need to check the indeterminate condition there too...

**Comment by Dalimil** [comment]
I have moved the 'checked' property from the `ref` code and also hopefully fixed some of the flickering by putting the `breakpointLoading` condition as a toggle for the `disabled` state of the input (instead of returning some kind of a new default checkbox, it just greys out the existing in that case).
out

**Comment by jasonLaster** [comment]
I pushed a couple of changes. Unfortunately they're amended to your flickering commit.

* updated checked logic so that we don't show the checkbox as checked when there aren't any breakpoints
* updated the `if` to just be `if (input)`

**Pull request merged**